enhancements/console/customize-add-page.md

```diff
@@ -0,0 +1,171 @@
+---
+title: customize-add-page
+authors:
+  - "@jerolimov"
+reviewers:
+  - "@spadgett"
+  - "@cvogt"
+  - "@jhadvig"
+approvers:
+  - "@spadgett"
+  - "@cvogt"
+creation-date: 2021-02-25
+last-updated: 2021-04-07
+status: implementable
+---
+
+# Customize Add Page
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Motivation
+
+The OpenShift Developer Console provides many ways to import applications and services. Customers who use only some of these action want to hide the unused actions from the add page.
+
+## Summary
+
+Cluster admins want to customize the add page of the OpenShift Developer Console.
+
+This list contains a long list of actions, some of them depends on installed operators like the OpenShift Pipelines Operator, the OpenShift Serverless Operator, and others.
+
+![Current Add page screenshot](customize-add-page.png)
+
+### Goals
+
+- Add a customization parameter to allow cluster admins to hide actions from the add page.
+- Provide a list of available actions so that the admin can hide them.
+
+### Non-Goals
+
+- Introducing a UI to configure this without a yaml editor.
+
+## Proposal
+
+### Extending the Console CRD
+
+Similar to [Customize Developer Catalog Categories](./catalog-categories.md) we extend the existing `operator.openshift.io/v1` / `Console` CRD. It provides already a `spec.customization` area where we can hide actions from the add page.
+
+The cluster admin can disable actions by ID. This ensures that later installed operators can show new actions without requiring the admin enable them manually.
+
+A final YAML should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    brand: online
+    addPage:
+      disabledActions:
+      - devcatalog/databases
+      - devcatalog/helmcharts
+      - devcatalog/operatorbacked
+      - devcatalog/virtualmachinetemplates
+      - devcatalog/managedservices
+      - fromgit
+      - fromdevfile
+      - fromdockerfile
+      - containerimage
+      - importyaml
+      - uploadajar
+      - knative.openshift.io/eventsource
+      - knative.openshift.io/channel
+      - tekton.dev/pipelines
+  ...
+```
+
+### Provide a code snippet in YAML editor
+
+The admin needs to know the current available actions to hide them. We don't want to provide the defaults in the CR so that we can change these defaults later without migrating the customer data.
+
+To provide the default values close to the editor, the console will provide a code snippets in the sidebar of the YAML editor when editing the `Console` operator config. The snippets should include the IDs of all actions.
```

Comment:
We probably need more than a snippet to document these, particularly since a lot of admins use the CLI. Have we given thought about where else we can publish these? Maybe in the console customization topic in the OpenShift docs and the API docs for the property.

Have we considered a simple UI for admins to update them (even if not in 4.8)? Maybe a modal similar to what we have for table column customization.

Reply:
A UI is being considered in future however plans yet in 4.8 or 4.9. This may fall under the umbrella of the same mechanism that will serve customized dashboards.

I agree that we can work with docs to provide guidance on customizing this and other configurations similar to it.

Reply:
@spadgett @christianvogt Should we discuss this in slack and open a ticket in the epics for this? I also expect that we should do this for all 4 (!) customization options we will have in 4.8.

1. dev catalog categories
2. available cluster roles to assign on the project page (project access)
3. disable actions on the add page
4. disable quick starts

Reply:
I think we can move forward with this enhancement and provide docs.